src/LightXML.jl

```diff
@@ -22,12 +26,23 @@ module LightXML
     XMLDocument, version, encoding, compression, standalone, root,
     parse_file, parse_string, save_file, set_root, create_root
 
-    using Compat
-    include("clib.jl")
-    include("errors.jl")
+typealias Xchar UInt8
+typealias Xstr Ptr{Xchar}
+
+# opaque pointer type (do not dereference!) corresponding to xmlBufferPtr in C
+immutable xmlBuffer end
+typealias Xptr Ptr{xmlBuffer}
+
+# pre-condition: p is not null
+# (After tests, it seems that free in libc instead of xmlFree
+#  should be used here.)
+_xcopystr(p::Xstr) = (r = bytestring(p); Libc.free(p); r)
```

Comment:
Note that this could use `Base.pointer_to_string` in 0.4 to avoid making a copy, but that's not currently exported.